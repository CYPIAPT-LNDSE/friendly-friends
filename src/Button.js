import React, { Component } from 'react';

class Button extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <button className="button">{this.props.label}</button>
    );
  }
}

export default Button;
