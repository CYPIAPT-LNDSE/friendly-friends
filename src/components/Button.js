import React from 'react';

const obj = {
  stroke: (e) => console.log('stroke'),
  talk: (e) => console.log('talk'),
  feed: (e) => console.log('feed'),
  play: (e) => console.log('play')
}

const Button = (props) => {
  return (
    <button onClick={(e) => obj[props.label](e)} className="button">{props.label}</button>
  );
}

export default Button;
