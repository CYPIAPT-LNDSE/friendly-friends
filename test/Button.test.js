import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button.js';

test('Button renders correctly', () => {
  const tree = renderer.create(<Button label="hello"/>).toJSON();
  expect(tree).toMatchSnapshot();
});
