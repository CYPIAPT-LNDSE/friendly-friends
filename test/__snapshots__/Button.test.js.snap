exports[`test Button renders correctly 1`] = `
<button
  className="button">
  hello
</button>
`;
